src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -589,179 +591,41 @@ private TypeGenerationSpec ParseTypeGenerationSpec(TypeToGenerate typeToGenerate
                 {
                     classType = ClassType.Enum;
                 }
-                else if (type.GetCompatibleGenericBaseType(_knownSymbols.IAsyncEnumerableOfTType) is INamedTypeSymbol iasyncEnumerableType)
```

Comment:
The deleted code has been extracted into the `TryResolveCollectionType` method .